From afe4e9cea9c9a7ea476c05ec84f120244d484039 Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Mon, 10 Jun 2024 09:59:27 +0200
Subject: [PATCH 0922/1695] dts: stm32: define RIFSC as an access controller
 for stm32mp23 boards

RIFSC is a firewall controller. Add the access-controllers property to
all RIFSC sub-nodes. Also add the "simple-bus" compatible for backward
compatibility and "#access-controllers-cells" to the RIFSC node.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 753c7280735e96a52a9c9f918f880e5c4669b6c4 dts: stm32: define RIFSC as an access controller for stm32mp23 boards]
Change-Id: Id13c26abbe9347acee8155cd4ab34da9e4462a1e
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/386446
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/arch/arm/dts/stm32mp231.dtsi | 26 ++++++++++++++++++++++++--
 1 file changed, 24 insertions(+), 2 deletions(-)

diff --git a/core/arch/arm/dts/stm32mp231.dtsi b/core/arch/arm/dts/stm32mp231.dtsi
index 44ca8e414..0bdd1736d 100644
--- a/core/arch/arm/dts/stm32mp231.dtsi
+++ b/core/arch/arm/dts/stm32mp231.dtsi
@@ -9,6 +9,10 @@
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/regulator/st,stm32mp25-regulator.h>
 #include <dt-bindings/reset/stm32mp25-resets.h>
+#include <dt-bindings/soc/stm32mp25-rif.h>
+#include <dt-bindings/soc/stm32mp25-rifsc.h>
+#include <dt-bindings/soc/stm32mp25-risab.h>
+#include <dt-bindings/soc/stm32mp25-risaf.h>
 
 / {
 	#address-cells = <2>;
@@ -252,11 +256,12 @@
 			};
 		};
 
-		rifsc: rifsc@42080000 {
-			compatible = "st,stm32mp25-rifsc";
+		rifsc: bus@42080000 {
+			compatible = "st,stm32mp25-rifsc", "simple-bus";
 			reg = <0x42080000 0x1000>;
 			#address-cells = <1>;
 			#size-cells = <1>;
+			#access-controller-cells = <2>;
 
 			lptimer1: timer@40090000 {
 				compatible = "st,stm32mp25-lptimer";
@@ -266,6 +271,7 @@
 				clock-names = "mux";
 				#address-cells = <1>;
 				#size-cells = <0>;
+				access-controllers = <&rifsc STM32MP25_RIFSC_LPTIM1_ID 0>;
 				status = "disabled";
 
 				counter {
@@ -299,6 +305,7 @@
 				clock-names = "mux";
 				#address-cells = <1>;
 				#size-cells = <0>;
+				access-controllers = <&rifsc STM32MP25_RIFSC_LPTIM2_ID 0>;
 				status = "disabled";
 
 				counter {
@@ -329,6 +336,7 @@
 				reg = <0x400e0000 0x400>;
 				interrupts = <GIC_SPI 115 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_USART2>;
+				access-controllers = <&rifsc STM32MP25_RIFSC_USART2_ID 0>;
 				status = "disabled";
 			};
 
@@ -337,6 +345,7 @@
 				reg = <0x400f0000 0x400>;
 				interrupts = <GIC_SPI 116 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_USART3>;
+				access-controllers = <&rifsc STM32MP25_RIFSC_USART3_ID 0>;
 				status = "disabled";
 			};
 
@@ -345,6 +354,7 @@
 				reg = <0x40100000 0x400>;
 				interrupts = <GIC_SPI 126 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_UART4>;
+				access-controllers = <&rifsc STM32MP25_RIFSC_UART4_ID 0>;
 				status = "disabled";
 			};
 
@@ -353,6 +363,7 @@
 				reg = <0x40110000 0x400>;
 				interrupts = <GIC_SPI 127 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_UART5>;
+				access-controllers = <&rifsc STM32MP25_RIFSC_UART5_ID 0>;
 				status = "disabled";
 			};
 
@@ -364,6 +375,7 @@
 				#address-cells = <1>;
 				#size-cells = <0>;
 				i2c-analog-filter;
+				access-controllers = <&rifsc STM32MP25_RIFSC_I2C1_ID 0>;
 				status = "disabled";
 			};
 
@@ -387,6 +399,7 @@
 				#address-cells = <1>;
 				#size-cells = <0>;
 				i2c-analog-filter;
+				access-controllers = <&rifsc STM32MP25_RIFSC_I2C7_ID 0>;
 				status = "disabled";
 			};
 
@@ -395,6 +408,7 @@
 				reg = <0x40220000 0x400>;
 				interrupts = <GIC_SPI 136 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_USART6>;
+				access-controllers = <&rifsc STM32MP25_RIFSC_USART6_ID 0>;
 				status = "disabled";
 			};
 
@@ -403,6 +417,7 @@
 				reg = <0x40330000 0x400>;
 				interrupts = <GIC_SPI 114 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_USART1>;
+				access-controllers = <&rifsc STM32MP25_RIFSC_USART1_ID 0>;
 				status = "disabled";
 			};
 
@@ -411,6 +426,7 @@
 				reg = <0x40370000 0x400>;
 				interrupts = <GIC_SPI 148 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_UART7>;
+				access-controllers = <&rifsc STM32MP25_RIFSC_UART7_ID 0>;
 				status = "disabled";
 			};
 
@@ -420,6 +436,7 @@
 				interrupts = <GIC_SPI 145 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_BUS_HASH>;
 				resets = <&rcc HASH_R>;
+				access-controllers = <&rifsc STM32MP25_RIFSC_HASH_ID 0>;
 				status = "disabled";
 			};
 
@@ -429,6 +446,7 @@
 				clocks = <&clk_rcbsec>, <&rcc CK_BUS_RNG>;
 				clock-names = "rng_clk", "rng_hclk";
 				resets = <&rcc RNG_R>;
+				access-controllers = <&rifsc STM32MP25_RIFSC_RNG_ID 0>;
 			};
 
 			iwdg1: watchdog@44010000 {
@@ -437,6 +455,7 @@
 				interrupts = <GIC_SPI 2 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_BUS_IWDG1>, <&rcc LSI_CK>;
 				clock-names = "pclk", "lsi";
+				access-controllers = <&rifsc STM32MP25_RIFSC_IWDG1_ID 0>;
 				status = "disabled";
 			};
 
@@ -448,6 +467,7 @@
 				clock-names = "mux";
 				#address-cells = <1>;
 				#size-cells = <0>;
+				access-controllers = <&rifsc STM32MP25_RIFSC_LPTIM3_ID 0>;
 				status = "disabled";
 
 				counter {
@@ -481,6 +501,7 @@
 				clock-names = "mux";
 				#address-cells = <1>;
 				#size-cells = <0>;
+				access-controllers = <&rifsc STM32MP25_RIFSC_LPTIM4_ID 0>;
 				status = "disabled";
 
 				counter {
@@ -514,6 +535,7 @@
 				clock-names = "mux";
 				#address-cells = <1>;
 				#size-cells = <0>;
+				access-controllers = <&rifsc STM32MP25_RIFSC_LPTIM5_ID 0>;
 				status = "disabled";
 
 				counter {
-- 
2.47.2

